### DISCUSSION TITLE
How to pass arguments by reference from c++ to python with pybind11?

### DISCUSSION THREAD

**Discussion by leonleeann** [pybind/pybind11#3991] [Q&A]
I'm using **PyBind11** to run a Python **interpreter**, and I need to call a Python function in c++ with some pointer arguments.

It looks like that a argument being passed to Python side should be freed normally by the Python interpreter, instead of the c++ main program.
But this time my argument is a pointer to a **static** object, it should **NOT** be freed by anyone.

I know that `pybind11::return_value_policy::reference` can be used to prevent a returning result from being freed, but it is for a returning object, not for arguments.

How to code such a binding/calling?

**Comment by steve-lorimer** [comment]
Did you ever find an answer to this?

**Comment by ManuelSchneid3r** [comment]
Encountered this issue as well. Passing a pointer works well but references throw an exception. Any progress here?
